Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromDDD.cc

```diff
@@ -199,16 +232,19 @@ GEMChamber* GEMGeometryBuilderFromDDD::buildChamber(DDFilteredView& fv, GEMDetId
                                         << std::endl;
 
   DDBooleanSolid solid = (DDBooleanSolid)(fv.logicalPart().solid());
-  std::vector<double> dpar = solid.solidA().parameters();
+  std::vector<double> dpar = (detId.station() == 0) ? solid.parameters() : solid.solidA().parameters();
 
   double dy = geant_units::operators::convertMmToCm(dpar[0]);   //length is along local Y
   double dz = geant_units::operators::convertMmToCm(dpar[3]);   // thickness is long local Z
   double dx1 = geant_units::operators::convertMmToCm(dpar[4]);  // bottom width is along local X
   double dx2 = geant_units::operators::convertMmToCm(dpar[8]);  // top width is along local X
-  dpar = solid.solidB().parameters();
-  dz += geant_units::operators::convertMmToCm(dpar[3]);  // chamber thickness
 
-  bool isOdd = detId.chamber() % 2;
+  if (detId.station() != 0) {
+    dpar = solid.solidB().parameters();
+    dz += geant_units::operators::convertMmToCm(dpar[3]);  // chamber thickness
+  }
+
+  bool isOdd = (detId.station() == 0) ? false : detId.chamber() % 2;
```

Comment:
GEMDetId::minStationId0

alternatively, given the frequency of this check, you could define `bool ge0Station = detId.station()==GEMDetId::minStationId0;`